amazon-kinesis-client/src/main/java/software/amazon/kinesis/leases/ShardSyncTaskManager.java

```diff
@@ -51,6 +51,28 @@
     private final ExecutorService executorService;
     @NonNull
     private final MetricsFactory metricsFactory;
+    @NonNull
+    private final ShardSyncer shardSyncer;
+
+    public ShardSyncTaskManager(ShardDetector shardDetector, LeaseRefresher leaseRefresher, InitialPositionInStreamExtended initialPositionInStream,
+                                boolean cleanupLeasesUponShardCompletion, boolean ignoreUnexpectedChildShards, long shardSyncIdleTimeMillis,
+                                ExecutorService executorService, MetricsFactory metricsFactory) {
+        this(shardDetector, leaseRefresher, initialPositionInStream, cleanupLeasesUponShardCompletion, ignoreUnexpectedChildShards, shardSyncIdleTimeMillis, executorService, metricsFactory, new ShardSyncer());
```

Comment:
You should not depend on the `ShardSyncTaskManager` creating the `ShardSyncer`, instead handle the creation in `DynamoDBLeaseManagementFactory`.

Reply:
Modified.